From 9fbba92869687f7def0f447d8ee8379b14eed3bf Mon Sep 17 00:00:00 2001
From: Homura <egzozu.be.bas@gmail.com>
Date: Thu, 29 Jan 2026 23:43:54 +0300
Subject: [PATCH] Enable D3D9 support for `ANGLE_DEFAULT_PLATFORM`.

---
 src/libANGLE/Display.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/libANGLE/Display.cpp b/src/libANGLE/Display.cpp
index 72fb02c..5886c3d 100644
--- a/src/libANGLE/Display.cpp
+++ b/src/libANGLE/Display.cpp
@@ -314,6 +314,13 @@ EGLAttrib GetDisplayTypeFromEnvironment()
     }
 #endif
 
+#if defined(ANGLE_ENABLE_D3D9)
+    if (angleDefaultEnv == "d3d9")
+    {
+        return EGL_PLATFORM_ANGLE_TYPE_D3D9_ANGLE;
+    }
+#endif
+
 #if defined(ANGLE_ENABLE_D3D11)
     if (angleDefaultEnv == "d3d11")
     {
-- 
2.52.0

